prov/efa/configure.m4

```diff
@@ -45,195 +41,160 @@ AC_DEFUN([FI_EFA_CONFIGURE],[
-			AC_MSG_WARN([The EFA provider is not supported on 32-bit systems.])
-		])
-
-	AS_IF([test $efa_happy -eq 1 ], [$1], [$2])
+		AC_CHECK_DECL(EFADV_DEVICE_ATTR_CAPS_RDMA_WRITE,
+			[have_caps_rdma_write=1],
+			[have_caps_rdma_write=0],
+			[[#include <infiniband/efadv.h>]])
 
-	AS_IF([test $efadv_support_extended_cq -eq 1],
-		[FI_CHECK_PACKAGE([efadv],
-			[infiniband/efadv.h],
-			[efa],
-			[efadv_create_cq],
-			[-libverbs],
-			[$efa_PREFIX],
-			[$efa_LIBDIR],
-			[],
-			[efadv_support_extended_cq=0])
-	      ])
+		AC_CHECK_DECL([ibv_is_fork_initialized],
+			[have_ibv_is_fork_initialized=1],
+			[have_ibv_is_fork_initialized=0],
+			[[#include <infiniband/verbs.h>]])
 
-	AS_IF([test $efadv_support_extended_cq -eq 1],
-		[FI_CHECK_PACKAGE([efadv],
-			[infiniband/efadv.h],
-			[efa],
-			[efadv_cq_from_ibv_cq_ex],
-			[-libverbs],
-			[$efa_PREFIX],
-			[$efa_LIBDIR],
-			[],
-			[efadv_support_extended_cq=0])
-	      ])
+		AC_CHECK_DECL([IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES],
+			[efa_support_data_in_order_aligned_128_byte=1],
+			[efa_support_data_in_order_aligned_128_byte=0],
+			[[#include <infiniband/verbs.h>]])
 
-	# efadv_wc_read_sgid is a static inline function, which is
-	# required to use extended CQ for recovering peer address.
-	# Extended CQ should be disabled if it is not available.
-	AS_IF([test $efadv_support_extended_cq -eq 1],
-		[AC_EGREP_HEADER(
-			[int efadv_wc_read_sgid],
-			[infiniband/efadv.h],
-			[],
-			[efadv_support_extended_cq=0])
+		dnl Check for ibv_reg_dmabuf_mr() in libibverbs if built with synapseai support.
+		AS_IF([test x"$have_synapseai" = x"1"],[
+			AC_CHECK_DECL([ibv_reg_dmabuf_mr],
+				[],
+				[AC_MSG_ERROR(
+					[ibv_reg_dmabuf_mr is required by synapseai but not available
+					in the current rdma-core library. Please build libfabric with
+					rdma-core >= v34.0])],
+				[[#include <infiniband/verbs.h>]])
 		])
 
-	AS_IF([test $efadv_support_extended_cq -eq 1],
-		[AC_DEFINE([HAVE_EFADV_CQ_EX], [1], [EFA device support extensible CQ])],
-		[AC_DEFINE([HAVE_EFADV_CQ_EX], [0], [EFA device does not support extensible CQ])]
-		)
-
-	AM_CONDITIONAL([HAVE_EFADV_CQ_EX], [test $efadv_support_extended_cq -eq 1])
-
-	# IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES is the flag we need test,
-	# but it is flag. If it is not there, we should disable it.
-	AS_IF([test $efa_support_data_in_order_aligned_128_byte -eq 1],
-		[AC_EGREP_HEADER(
-			[IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES],
-			[infiniband/verbs.h],
+		dnl For efadv_support_extended_cq, we check several things,
+		dnl and if any of them fail, we disable CQ_EX
+		efadv_support_extended_cq=1
+		AC_CHECK_DECL([efadv_create_cq],
 			[],
-			[efa_support_data_in_order_aligned_128_byte=0])
-		])
+			[efadv_support_extended_cq=0],
+			[[#include <infiniband/efadv.h>]])
+		AC_CHECK_DECL([efadv_cq_from_ibv_cq_ex],
+			[],
+			[efadv_support_extended_cq=0],
+			[[#include <infiniband/efadv.h>]])
+		AC_CHECK_DECL([efadv_wc_read_sgid],
```

Comment:
I would let @wenduwan have a comment here. I think he uses EGREP_HEADER for some reason. Not sure AC_CHECK_DECL works here

Reply:
I checked the unit test report and the tests were run so I think it worked.